### PR TITLE
feat: EASIE documentation

### PR DESCRIPTION
**Pull request by mzhong9723**
> - https://clerk.com/docs/pr/1706/authentication/enterprise-connections/overview
> - https://clerk.com/docs/pr/1706/authentication/enterprise-connections/account-linking
> - https://clerk.com/docs/pr/1706/authentication/enterprise-connections/authentication-flows
> - https://clerk.com/docs/pr/1706/authentication/enterprise-connections/easie/google
> - https://clerk.com/docs/pr/1706/authentication/enterprise-connections/easie/microsoft

### Explanation:

Documentation update for EASIE connections

### This PR:

- Refactors documentation that applies to enterprise connections as a whole to reference "Enterprise SSO" instead of "SAML"
- Adds new EASIE-specific documentation

### CODE REVIEW

**Comment by mzhong9723** [comment]
There are some inconsistencies in capitalization for certain terms:
- Enterprise Connection
- Enterprise Account
- Identity Provider

I have it on my TODOs to fix this!

File: docs/authentication/saml/overview.mdx [hunk 1]
```diff
@@ -1,52 +1,69 @@
 ---
```
Comment (mzhong9723) [inline docs/authentication/saml/overview.mdx:1]: TODO: Add some specific documentation about how to set both of these up in the dashboard

File: docs/authentication/enterprise-connections/easie/google.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  ### Set up an enterprise connection in Clerk
+
+  1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections).
+  1. In the top navigation, select **Configure**. Then in the sidebar, select **SSO Connections**.
+  1. Select the **Add connection** button, and select **For specific domains**.
+  1. Below EASIE, select **Google** as the identity provider.
+  1. Add the **Specific Domain** that you want to allow this connection for. This is the domain of the users you want to allow to sign in to your application.
+  1. Ensure that **Use custom credentials** is toggled on.
+
+  ### Create a Google Developer project
```
Comment (LauraBeatris) [inline docs/authentication/enterprise-connections/easie/google.mdx:54]: I think here we could mention regarding the ability to re-use the credentials of an existing social connection.

**Review by izaaklauer — APPROVED** [review]
We'll want to add a section about deprovisioning when those features land, but I think everything in here is great.

**Review by LauraBeatris — DISMISSED** [review]
Blocking merging this PR while team docs is working on it

File: docs/authentication/enterprise-connections/account-linking.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+By default, Clerk requires email verification at sign-up. For instances that have disabled this behavior, there is a possibility that an account may be created using an unverified email address.
+
+To configure email verification at sign-up:
+
+1. In the Clerk Dashboard, navigate to the [**Email, Phone, Username**](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username) page.
+1. Next to **Email address**, select the settings icon and uncheck the **Verify at sign-up** toggle.
+
+When a user signs into your app using an IdP, Clerk automatically links the Enterprise account to the existing account by **also verifying the existing email address** and signing the user in. This includes accounts protected by passwords, as the Enterprise sign-in flow automatically bypasses password verification.
+
+### Email is not verified
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/account-linking.mdx:32]: Is this what happens when "Email is not verified and verification is required" ?
If verification is required, and an email is not verified, then Clerk creates a completely new account?

File: docs/authentication/enterprise-connections/account-linking.mdx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+### Email is not verified and verification isn't required
+
+By default, Clerk requires email verification at sign-up. For instances that have disabled this behavior, there is a possibility that an account may be created using an unverified email address.
+
+To configure email verification at sign-up:
+
+1. In the Clerk Dashboard, navigate to the [**Email, Phone, Username**](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username) page.
+1. Next to **Email address**, select the settings icon and uncheck the **Verify at sign-up** toggle.
+
+When a user signs into your app using an IdP, Clerk automatically links the Enterprise account to the existing account by **also verifying the existing email address** and signing the user in. This includes accounts protected by passwords, as the Enterprise sign-in flow automatically bypasses password verification.
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/account-linking.mdx:30]: So if verification isn't required, and the email isn't verified, Clerk verifies the email for them, and then signs the user in? Is this correct? I just want to make sure our docs are still accurate.

File: docs/authentication/enterprise-connections/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+## EASIE
+
+[EASIE](https://easie.dev) SSO is a way for applications to provide enterprise-grade SSO through a multi-tenant OpenID provider. It is designed to be an easier alternative to SAML SSO.
+
+The following IdPs are supported: Google Workspace and Microsoft Entra ID. For _development instances_, Clerk uses preconfigured shared credentials and redirect URIs—no other configuration is needed. For _production instances_, you must provide custom credentials. Follow the steps outlined in the guides to complete the setup:
+
+- [Google](docs/authentication/social-connections/google)
+- [Microsoft](docs/authentication/social-connections/azure)
+
+### Automatic Deprovisioning
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/overview.mdx:36]:
```suggestion
### Automatic deprovisioning
```

File: docs/authentication/enterprise-connections/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Clerk prevents users linked to deprovisioned accounts in the OpenID provider from accessing the application.
+
+Within 10 minutes of a user being removed from the OpenID provider (e.g. [suspendeded](https://support.google.com/a/answer/33312?hl=en) or [deleted](https://support.google.com/a/answer/33314?hl=en) via Google Workspace, or [deleted](https://learn.microsoft.com/en-us/entra/fundamentals/how-to-create-delete-users#delete-a-user) via Microsoft Entra), Clerk will recognize that the user has been deprovisioned and will revoke that user's existing sessions.
+
+It is ultimately the application's responsibility to handle this unauthenticated state and display something appropriate to the user. For example, NextJS applications using [`auth.Protect`](docs/references/nextjs/auth#auth-protect) will automatically redirect the user to the sign-in page.
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/overview.mdx:42]:
```suggestion
It is ultimately the application's responsibility to handle this unauthenticated state and display something appropriate to the user. For example, Next.js applications using [`auth.protect()`](docs/references/nextjs/auth#auth-protect) will automatically redirect the user to the sign-in page.
```

File: docs/authentication/enterprise-connections/account-linking.mdx (hunk 4 above)
Comment (izaaklauer) [reply docs/authentication/enterprise-connections/account-linking.mdx]: That's correct! I just tested it to make sure.

File: docs/authentication/enterprise-connections/account-linking.mdx (hunk 3 above)
Comment (izaaklauer) [reply docs/authentication/enterprise-connections/account-linking.mdx:31]: This is a great question. This lead @mzhong9723 and I down a bit of a rabbit hole testing this.

To answer your question:
- WHEN a user has an existing unverified email address (the account was created with raw email/password with "verify at sign-up" turned off)
- AND the application owner configured EASIE **or** SAML for a domain matching the user's existing email address
- AND the user signs in, and gets redirected to the IDP
- THEN that account linking occurs. We **do not** create a new account, but instead update the existing account's email address to be verified.

We suspect this is a reversion, and it's not EASIE-specific. We're tracking it here: https://linear.app/clerk/issue/ORGS-361/handle-preventing-account-linking-to-unverified-email-addresses

File: docs/authentication/enterprise-connections/easie/microsoft.mdx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+<Steps>
+  ### Enable Microsoft as an EASIE connection
+
+  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
+  1. Select **Add connection** and select **For specific domains**.
+  1. Under **EASIE**, select **Microsoft** as the identity provider.
+  1. Add the **Specific Domain** that you want to allow this connection for. This is the domain of the users you want to allow to sign in to your app.
+  1. Select **Add connection**.
+  1. Ensure that **Use custom credentials** is toggled on.
+  1. Save the **Redirect URI** somewhere secure.
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/easie/microsoft.mdx:58]:
```suggestion
  1. Ensure that **Use custom credentials** is toggled on.
  1. Save the **Redirect URI** somewhere secure. Keep this page open.
```

File: docs/authentication/enterprise-connections/easie/microsoft.mdx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+     1. Under **Supported account types**, select **Accounts in any organizational directory (Any Microsoft Entra ID tenant – Multitenant) and personal Microsoft accounts (e.g. Skype, Xbox)**.
+     1. Under **Redirect URI (Optional)**, select **Web**.
+     1. Select **Register** to submit the form.
+
+  ### Get your client ID and client secret
+
+  Once your Microsoft Entra ID app is created, or once you select your app from the Microsoft Azure portal, you'll be redirected to its **Overview**.
+
+  1. From your app's overview, save the **Application (client) ID** somewhere secure. You'll need it to connect your Microsoft Entra ID app to your Clerk app.
+  1. Under **Client credentials**, select **Add a certificate or secret** to generate a Client Secret. You'll be redirected to the **Certificate & secrets** page.
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/easie/microsoft.mdx:79]:
```suggestion
  1. Under **Client credentials**, select **Add a certificate or secret** to generate a **Client Secret**. You'll be redirected to the **Certificate & secrets** page.
```

File: docs/authentication/enterprise-connections/easie/google.mdx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+<Steps>
+  ### Enable Google as an EASIE connection
+
+  1. In the Clerk Dashboard, navigate to the [**SSO Connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
+  1. Select **Add connection** and select **For specific domains**.
+  1. Below EASIE, select **Google** as the identity provider.
+  1. Add the **Specific Domain** that you want to allow this connection for. This is the domain of the users you want to allow to sign in to your application.
+  1. Select **Add connection**.
+  1. Ensure that **Use custom credentials** is toggled on.
+  1. Save the **Redirect URI** somewhere secure.
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/easie/google.mdx:58]:
```suggestion
  1. Ensure that **Use custom credentials** is toggled on.
  1. Save the **Redirect URI** somewhere secure. Keep this page open.
```

File: docs/authentication/enterprise-connections/easie/google.mdx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  ### Create a Google Developer project
+
+  1. Navigate to the [Google Cloud Console](https://console.cloud.google.com/).
+  1. Select a project or [create a new one](https://console.cloud.google.com/projectcreate).
+  1. If the **APIs & Services** page isn't already open, open the menu on the left and select **APIs & Services**.
+  1. In the menu on the left, select **Credentials**.
+  1. Select **Create Credentials**. Then, select **OAuth client ID.** You may need to [configure your OAuth consent screen](https://support.google.com/cloud/answer/6158849?hl=en#userconsent\&zippy=%2Cuser-consent).
+  1. Select the appropriate application type for your project. Most likely, you will choose **Web application**.
+  1. In the **Authorized redirect URIs** setting, paste the **Redirect URI** value you saved from the Clerk Dashboard.
+  1. Select **Create**. Keep this page open.
```
Comment (alexisintech) [inline docs/authentication/enterprise-connections/easie/google.mdx:69]:
```suggestion
  1. Select **Create**.
```

**Review by LauraBeatris — APPROVED** [review]

**Pull request merged**